### PR TITLE
Add support for configuration of connection timeout and read timeout in ...

### PR DESCRIPTION
**Pull request by mariuszs**
...ServiceRestClient

### CODE REVIEW

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientConfiguration.groovy [hunk 1]
```diff
@@ -35,6 +37,9 @@ import java.util.concurrent.ThreadFactory
 @EnableServiceDiscovery
 class ServiceRestClientConfiguration {
 
+    @Value('${microservice.restclient.connectionTimeout:-1}') int connectionTimeoutMillis
```
Comment (nurkiewicz) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientConfiguration.groovy]: Maybe we do want to have some liberal (like 1 minute) but still hard-coded timeout? Not sure...

Comment (mariuszs) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientConfiguration.groovy]: Maybe, default was -1 and I dont want to change this behaviour with my change.

Comment (nurkiewicz) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientConfiguration.groovy]: OK, sounds fair.

One more thing, [here](https://github.com/4finance/micro-infra-spring/pull/175/files#diff-6cc9c8bfe82f8c4ebddf18c337e1e040R40) we introduce different naming convention for `ServiceRestClient` parameters. It's compatible with `rest.api...` Swagger config properties. Do you mind if I change this to `rest.client.connectionTimeout`?

File: micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientIntegrationSpec.groovy [hunk 2]
```diff
@@ -38,5 +48,24 @@ class ServiceRestClientIntegrationSpec extends MvcWiremockIntegrationSpec {
         then:
             result.body == CONTEXT_SPECIFIC_FOOBAR
     }
-    
+
+    def "should throw an exception when service does not respond"() {
+        given:
+            stubInteraction(wireMockGet('/delayed'), aResponse()
+                    .withFixedDelay(readTimeoutMillis * 2)
+                    .withBody("THIS SHOULD TIME OUT")
```
Comment (nurkiewicz) [inline micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientIntegrationSpec.groovy]: Is body necessary here? It's never gonna be used anyway

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientConfiguration.groovy (hunk 1 above)
Comment (mariuszs) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientConfiguration.groovy]: No, please change this.

File: micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientIntegrationSpec.groovy (hunk 2 above)
Comment (mariuszs) [inline micro-infra-spring-base/src/test/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientIntegrationSpec.groovy]: No, please remove it

**Pull request merged**